src/mscorlib/shared/System/TimeZoneInfo.cs

```diff
@@ -1972,11 +1966,7 @@ private static void ValidateTimeZoneInfo(string id, TimeSpan baseUtcOffset, Adju
                         throw new InvalidTimeZoneException(SR.Argument_AdjustmentRulesNoNulls);
                     }
 
-                    // FUTURE: check to see if this rule supports Daylight Saving Time
-                    // adjustmentRulesSupportDst = adjustmentRulesSupportDst || current.SupportsDaylightSavingTime;
-                    // FUTURE: test baseUtcOffset + current.StandardDelta
-
-                    if (UtcOffsetOutOfRange(baseUtcOffset + current.DaylightDelta))
+                    if (!IsValidAdjustmentRuleOffest(baseUtcOffset, current))
```

Comment:
Typo IsValidAdjustmentRuleOffest -> IsValidAdjustmentRuleOffset

Reply:
nice catch - I couldn't see the typo for couple of seconds even after you pointed it 😄